sdk/runtime/runner.go

```diff
@@ -156,7 +156,11 @@ func setupMetrics(ctx context.Context, runenv *RunEnv) (doneCh chan error) {
 				if err != nil {
 					continue
 				}
-				_, _ = resp.Body.Read(b)
+				_, err = io.ReadFull(resp.Body, b)
+				resp.Body.Close()
+				if err != nil {
```

Comment:
I think this was breaking the functionality for some weird reason. I remember applying it, and then we had to revert.

Reply:
It was breaking because we were not properly managing `io.EOF`. `io.ReadFull` does that for us, so this patch shouldn't cause a regression.